lwt/examples/resume_echo_server.ml

```diff
@@ -0,0 +1,107 @@
+open Lwt
+open Ex_common
+
+let string_of_unix_err err f p =
+  Printf.sprintf "Unix_error (%s, %s, %s)"
+    (Unix.error_message err) f p
+
+
+module HT = Hashtbl.Make (Tls.Core.PreSharedKeyID)
+let cache_psk, psk_cache =
+  let cache = HT.create 7 in
+  ((fun ed -> HT.add cache ed.Tls.Core.psk_id ed),
+   (fun id -> if HT.mem cache id then Some (HT.find cache id) else None))
```

Comment:
As noted previously I think we should be careful about what implementations we provide for example session caches, since chances are people will copy-paste them if we don't provide a default/easy-to-use one in the library.
Thus, I believe our example session caches should be some sort of fixed-size LRU to prevent running out of memory.

Reply:
as noted above, I think this is a valid feature request, but just not in scope for this PR.